actionpack/lib/action_dispatch/journey/router.rb

```diff
@@ -122,6 +122,7 @@ def find_routes req
 
         def match_head_routes(routes, req)
           head_routes = match_routes(routes, req)
+          head_routes.delete_if { |route| route.verb == // }
```

Comment:
I don't think that the responsibility of rejecting routes without a `request_method` should be handled by `match_head_routes`. [`AD::Journey::Route#matches`](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/journey/route.rb#L93) should reject routes which does not have a `request_method` constraint.

cc @pixeltrix WDYT?